finetune/target_encoders.py

```diff
@@ -68,7 +68,58 @@ def transform(self, y):
         labels = super().transform(y)
         return self._make_one_hot(labels)
 
+class OrdinalRegressionEncoder(OrdinalEncoder, BaseEncoder):
+    
+    def __init__(self):
+        self.num_outputs = None
+        super().__init__()
+        
+    def fit(self, x):
+        x = np.array(x)
+        rank = len(x.shape)
+        if rank == 1:
+            x = np.expand_dims(x, 1)
+        self.fit_transform(x)
+        return self
+
+    def transform(self, x):
+        x = np.array(x)
+        rank = len(x.shape)
+        if rank == 1:
+            x = np.expand_dims(x, 1)  # for single output value regression.
+        labels = super().transform(x)
+        labels = self.rank_to_thresholds(labels)
+        return labels
+        
+    def fit_transform(self, x):
+        super().fit(x)
+        labels = self.transform(x)
+        self.num_outputs = labels.shape[1]
+        return labels
+    
+    def rank_to_thresholds(self,x):
+        #changes a one-variable rank into an array of 1s and 0s defining the target output of each threshold
+        x = x[:10000]
+        num_thresholds = len(self.categories_[0])-1
+        thresholds = [np.concatenate((np.ones(int(rank)),np.zeros((num_thresholds-int(rank))))) for rank in x]
```

Comment:
spacing

Reply:
more idiomatic to construct matrix ahead of time (as all zeros) and fill in with ones